File: 0268-missing-number/0268-missing-number.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        map<int,int> mp;
        int missingNumber = -1;
        for(int i = 0; i < nums.size(); i++) mp[i] = 0;
        for(int num: nums) mp[num]++;
        for(auto [key,val]: mp) {
            if(val == 0) missingNumber = key;
        }

        if(missingNumber == -1) {
            return nums.size();
        }
        return missingNumber;
    }
};

File: 0268-missing-number/0268-missing-number.js
/**
 * @param {number[]} nums
 * @return {number}
 */

 function binarySearch(arr) {
    let left = 0,right = arr.length-1;
    while(left <= right) {
        let mid = Math.floor(left + (right - left) / 2);
        if(arr[mid] > mid) right = mid-1;
        else left = mid+1;
    }
    return left;
 }
var missingNumber = function(nums) {
    nums.sort((a,b) => a-b);
    return binarySearch(nums);
};